### PR TITLE
Reduce if statement as it has the same expression on both sides of '&&'

### PR DESCRIPTION
**Pull request by danbey**
If statement has the same expression on both sides so this can be reduce

Signed-off-by: Dan Ben Yosef <danbey@gmail.com>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by danbey** [comment]
please link this issue to the PR : https://github.com/opencv/opencv/issues/19524

File: 3rdparty/carotene/src/resize.cpp [hunk 1]
```diff
@@ -758,7 +758,7 @@ inline void resizeAreaRounding(const Size2D &ssize, const Size2D &dsize,
     }
     else if (channels == 3)
     {
-        if ((wr == 2.0f) && (wr == 2.0f))
```
Comment (asmorkalov) [inline 3rdparty/carotene/src/resize.cpp:761]: According to `isResizeAreaSupported` that is called before hand to check HAL kernel applicability the condition should be `if ((wr == 2.0f) && (hr == 2.0f))`.

**Review by asmorkalov — CHANGES_REQUESTED** [review]
Thanks for the contribution! The same bug exists in branch 3.4 too. This patch should [go](https://github.com/opencv/opencv/wiki/How_to_contribute#faq) into [3.4 branch](https://github.com/opencv/opencv/wiki/Branches) first. We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

So, please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no need to re-open PR, apply changes "inplace".

**Review by asmorkalov — APPROVED** [review]
Well done!

**Pull request merged**